src/mscorlib/src/System/TimeZoneInfo.cs

```diff
@@ -2178,9 +2179,93 @@ static public TimeZoneInfo FindSystemTimeZoneById(string id)
             return result;
         }
 
-        static public ReadOnlyCollection<TimeZoneInfo> GetSystemTimeZones() {
-            // UNIXTODO
-            throw new NotImplementedException();
+        /// <summary>
+        /// Returns a collection of TimeZoneInfo instances containing all valid TimeZones
+        /// from the local machine.  The entries in the collection are sorted by
+        /// 'DisplayName'.
+        ///
+        /// This method does *not* throw TimeZoneNotFoundException or
+        /// InvalidTimeZoneException, but it can throw a SecurityException.
+        /// </summary>
+        static public ReadOnlyCollection<TimeZoneInfo> GetSystemTimeZones()
+        {
+            // UNIXTODO: Issue #2497 use CachedData
+
+            List<TimeZoneInfo> timeZoneInfos = new List<TimeZoneInfo>();
+
+            string timeZoneDirectory = GetTimeZoneDirectory();
+            foreach (string timeZoneId in GetTimeZoneIds(timeZoneDirectory))
+            {
+                if (!string.IsNullOrEmpty(timeZoneId))
+                {
+                    try
+                    {
+                        TimeZoneInfo timeZoneInfo = FindSystemTimeZoneById(timeZoneId);
+                        timeZoneInfos.Add(timeZoneInfo);
+                    }
+                    catch (InvalidTimeZoneException) { }
+                    catch (TimeZoneNotFoundException) { }
+                }
+            }
+
+            // sort and copy the TimeZoneInfo's into a ReadOnlyCollection for the user
+            timeZoneInfos.Sort(new TimeZoneInfoComparer());
```

Comment:
There is a method `ClearCachedData()` that is public in the code, but there doesn't appear to be a .NET Core Contract for it.
Also, there is something of an Easter egg in the `ConvertTime(DateTime, TimeZoneInfo)` method:

```C#
        static public DateTime ConvertTime(DateTime dateTime, TimeZoneInfo destinationTimeZone) {
            if (destinationTimeZone == null) {
                throw new ArgumentNullException("destinationTimeZone");
            }
            Contract.EndContractBlock();

            // Special case to give a way clearing the cache without exposing ClearCachedData()
            if (dateTime.Ticks == 0) {
                ClearCachedData();
            }
```

So people could clear the cache those 2 ways, if they are concerned with the TZ data being out of date while the app is running.